metadata.pl

```diff
@@ -0,0 +1,273 @@
+    if ($keys ne "null") {
+        if ( exists($properties{$keys})) {
+            print $properties{$keys};
+        } else {
+            print "null";
+        } 
+        exit;
+    }
+    # convert the run.properties file to scenario.json, creating json arrays
+    # and subarrays in the appropriate places 
+    if ( $jsonify ) {
+        foreach my $pp (@paren_properties) {
+            if ( exists($properties{$pp})) {
+                my @list_items = split(" ",$properties{$pp});
+                # remove leading and trailing parentheses, which will be the first and last fields
+                if ( $list_items[-1] eq ")" ) {
+                    pop(@list_items);
+                }
+                if ( $list_items[0] eq "(" ) {
+                    shift(@list_items);
+                }
+                # if the list is empty, add the list item "null"
+                if ( @list_items == 0 ) {
+                    push(@list_items,"null");
+                }
+                $properties{$pp} = \@list_items; # add list to hash to replace scalar representation of this list
+            }             
+        }
+        # turn property values that are meant to be a list (comma separated)
+        # into a json array
+        foreach my $cp (@comma_properties) {
+            if ( exists($properties{$cp})) {
+                my @list_items = split(",",$properties{$cp});
+                $properties{$cp} = \@list_items; # add list to hash to replace scalar representation of this list
+            }             
+        }
+        # turn the output file properties into a sublist
+        # with their own hashes for file name, file format, etc
+        my @outputlist; 
+        foreach my $op (@outputfile_properties) {
+            if ( exists($properties{"$op File Name"}) ) {
+                # add the $op as the new key
+                # populate it with a hash containing File Name and Format
+                # delete the old "$op File Name" and "$op Format" keys
+                my $fn = $properties{"$op File Name"};
+                my $ff = $properties{"$op Format"};
+                delete $properties{"$op File Name"};
+                delete $properties{"$op Format"};
+                my %fp = ( "description", $op, "name", $fn, "format", $ff );
+                push(@outputlist,\%fp);
+            }
+        } 
+        $properties{"adcirc.files.output"} = \@outputlist;
+        # now encode as json and write out
+        unless ( open(SJ,">$dirpath/scenario.json") ) {
+            &stderrMessage("ERROR","Could not open '$dirpath/scenario.json' for writing: $!.");
+        }
+        my $json = JSON::PP->new->utf8->pretty->canonical->encode(\%properties);
+        print SJ $json;
+        close(SJ);
+    }
+} else {
+    &stderrMessage("ERROR","File type was not recognized (requires either 'json' or 'yaml' or '.properties').");
+}
+
+1;
+
+#
+# write a log message to stderr
+sub stderrMessage () {
```

Comment:
@jasonfleming - next time might want to look into `Log::Log4perl` - it's not currently installed by default, but would not be a bad addition to our set of modules. It makes logging with multiple output targets a little more manageable.